plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-system-status-v2-controller.php

```diff
@@ -726,9 +723,8 @@ public function get_environment_info_per_fields( $fields ) {
 		}
 
 		// WP memory limit.
-		$wp_memory_limit = wc_let_to_num( WP_MEMORY_LIMIT );
-		if ( function_exists( 'memory_get_usage' ) ) {
-			$wp_memory_limit = max( $wp_memory_limit, wc_let_to_num( @ini_get( 'memory_limit' ) ) ); // phpcs:ignore WordPress.PHP.NoSilencedErrors.Discouraged
+		if ( class_exists('WP_Site_Health') ) {
+			$wp_memory_limit = WP_Site_Health::get_instance()->php_memory_limit;
```

Comment:
With the old code and memory limit configured as -1 in `php.ini`, I see that with the existing code the status report is showing a memory limit of 256M (the value of `WP_MEMORY_LIMIT`), but with your change it shows just -1. Is this expected?